File: app/src/main/java/com/doubean/ford/ui/groups/groupTab/GroupTabViewModelFactory.java
package com.doubean.ford.ui.groups.groupTab;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.doubean.ford.data.repository.GroupRepository;
import com.doubean.ford.data.repository.GroupsFollowsAndSavesRepository;

public class GroupTabViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private GroupRepository repository;
    private String groupId;
    private String tagId;
    GroupsFollowsAndSavesRepository groupsFollowsAndSavesRepository;

    public GroupTabViewModelFactory(@NonNull GroupRepository groupRepository, GroupsFollowsAndSavesRepository groupsFollowsAndSavesRepository, String groupId, String tagId) {
        super();
        this.repository = groupRepository;
        this.groupsFollowsAndSavesRepository = groupsFollowsAndSavesRepository;
        this.groupId = groupId;
        this.tagId = tagId;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {

        return (T) new GroupTabViewModel(repository, groupsFollowsAndSavesRepository, groupId, tagId);
    }
}

File: app/src/main/java/com/doubean/ford/ui/groups/groupTab/GroupTabViewModelFactory.kt
package com.doubean.ford.ui.groups.groupTab

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.doubean.ford.data.repository.GroupRepository
import com.doubean.ford.data.repository.GroupsFollowsAndSavesRepository

class GroupTabViewModelFactory(
    private val repository: GroupRepository,
    var groupsFollowsAndSavesRepository: GroupsFollowsAndSavesRepository,
    private val groupId: String,
    private val tagId: String?
) : ViewModelProvider.NewInstanceFactory() {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return GroupTabViewModel(repository, groupsFollowsAndSavesRepository, groupId, tagId) as T
    }
}